lib/vrl/core/src/target.rs

```diff
@@ -51,4 +51,12 @@ pub trait Target: std::fmt::Debug {
     /// If `compact` is true, after deletion, if an empty object or array is
     /// left behind, it should be removed as well, cascading up to the root.
     fn remove(&mut self, path: &LookupBuf, compact: bool) -> Result<Option<Value>, String>;
+
+    fn get_metadata(&self, _key: &str) -> Result<Option<vrl_core::Value>, String> {
+        Err("metadata not available".to_string())
+    }
+
+    fn set_metadata(&mut self, _key: &str, _value: String) -> Result<(), String> {
```

Comment:
Do we need `remove_metadata` here too?

Reply:
That target file is not actually used. I'll delete it.